test/config/test/project.test.ts

```diff
@@ -7,6 +7,9 @@ test.each([
   { pattern: '*j*', expected: ['project_1', 'project_2'] },
   { pattern: 'project*', expected: ['project_1', 'project_2'] },
   { pattern: 'space*', expected: ['space_1'] },
+  { pattern: '!project_1', expected: ['project_2', 'space_1'] },
+  { pattern: '!project*', expected: ['space_1'] },
+  { pattern: '!project', expected: ['project_1', 'project_2', 'space_1'] },
 ])('should match projects correctly: $pattern', async ({ pattern, expected }) => {
```

Comment:
Hm, this test seems a bit flawed. What about adding something like this?

```ts
expect(ctx.projects.map(p => p.name)).toEqual(expected)
```

Note: `await runVitest` returns `ctx`